sfge/src/test/java/com/salesforce/rules/unusedmethod/ConstructorsTest.java

```diff
@@ -149,28 +137,50 @@ public void constructorUncalledByGrandchildSuper_expectViolation(Integer arity)
         assertViolations(sourceCodes, assertion);
     }
 
+    /**
+     * Simple tests for cases where one class's constructor is called somewhere in another class.
+     *
+     * @param arity - The arity of the tested constructor.
+     */
+    @ValueSource(ints = {0, 1})
+    @ParameterizedTest(name = "{displayName}: Arity {0}")
+    public void constructorCalledViaNew_expectNoViolation(Integer arity) {
+        // spotless:off
+        String[] sourceCodes = new String[]{
+            // Declare our tested class.
+            "global class TestedClass {\n"
+            // Declare a constructor with the expected arity that does nothing in particular.
+          + "    public TestedClass(" + StringUtils.repeat("boolean b", arity) + ") {}\n"
+          + "}",
+            // Declare a class to invoke the tested constructor.
+            "global class InvokerClass {\n"
+            // Annotate the invoker method to avoid tripping the rule.
+          + "    /* sfge-disable-stack UnusedMethodRule */\n"
+          + "    public static TestedClass invokeTestedConstructor() {\n"
+            // Invoke the tested constructor with however many parameters it expects.
+          + "        return new TestedClass(" + StringUtils.repeat("true", arity) + ");\n"
+          + "    }\n"
+          + "}"
+        };
+        // spotless:on
+        assertNoViolations(sourceCodes, 1);
+    }
+
     /**
      * Tests for cases where an inner class's constructor is called in the outer class.
      *
-     * @param variable - The way the variable is declared.
      * @param constructor - The way the constructor is invoked.
      */
-    @CsvSource({
-        "OuterClass.InnerClass, OuterClass.InnerClass",
-        "OuterClass.InnerClass, InnerClass",
-        "InnerClass, OuterClass.InnerClass",
-        "InnerClass, InnerClass"
-    })
-    @ParameterizedTest(name = "{displayName}: Declared as {0}, constructed as new {1}()")
-    @Disabled
-    public void innerConstructorUsedByOuter_expectNoViolation(String variable, String constructor) {
+    @ValueSource(strings = {"OuterClass.InnerClass", "InnerClass"})
+    @ParameterizedTest(name = "{displayName}: OuterClass.InnerClass constructed as new {0}()")
```

Comment:
The syntax with which the variable was declared ended up not impacting how the rule executes, so I removed the ability to configure that and cut the number of cases.

Reply:
How about we leave the initial set of ways of calling so that we'll know if there's ever a regression while making changes?